build/bazel/remote/execution/v2/remote_execution.proto

```diff
@@ -484,6 +484,12 @@ message Action {
   // timeout is too short. By encoding it directly in the `Action`, a lower
   // timeout will result in a cache miss and the execution timeout will fail
   // immediately, rather than whenever the cache entry gets evicted.
+  //
+  // The timeout does not necessarily correspond to elapsed real time. A
```

Comment:
It seems we didn't really detail the purpose of the timeout at all. How about expanding the start of this paragraph to cover it at the same time, e.g.

```
The timeout is only intended to cover the "execution" of the specified action and not time in queue
nor any overheads before or after execution such as marshalling inputs/outputs. The server
SHOULD avoid including time spent the client doesn't have control over, and MAY extend the
timeout to account for delays that occur during execution itself (e.g., lazily loading data from the
Content Addressable Storage, live migration of virtual machines, emulation overhead).
```

?

Reply:
That sounds good!

That said, I just realised that there is also another small feature I'd like to add as part of this PR. Now that workers get a lot of freedom in applying the timeout, it would also be useful if the amount of time as measured by the worker is provided as part of the results. We currently only offer a start and end time, but those can no longer be used to compute an execution duration that is usable as a timeout value.

What are your thoughts on the latest commit I just pushed?

Reply:
Generally, makes sense to add a field for this as well, so SGTM. I'd personally prefer the semantics be listed on the `timeout` field and have the `execution_duration` reference it, rather than the other way around, since `timeout` is the input parameter clients should be sending whereas `execution_duration` is the optional output metadata. But that's mostly about the place the comment should live, semantics make sense either way :).

Also, I'd still like to see the timeout comment expanded to cover the purpose overall, as per my previous comment.